include/nbl/builtin/hlsl/bitonic_sort/common.hlsl

```diff
@@ -0,0 +1,53 @@
+#ifndef _NBL_BUILTIN_HLSL_BITONIC_SORT_COMMON_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BITONIC_SORT_COMMON_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/concepts.hlsl>
+#include <nbl/builtin/hlsl/math/intutil.hlsl>
+#include <nbl/builtin/hlsl/utility.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace bitonic_sort
+{
+
+template<typename KeyType, typename ValueType, typename Comparator>
+void compareExchangeWithPartner(
+	bool takeLarger,
+	NBL_REF_ARG(pair<KeyType, ValueType>) loPair,
+	NBL_CONST_REF_ARG(pair<KeyType, ValueType>) partnerLoPair,
+	NBL_REF_ARG(pair<KeyType, ValueType>) hiPair,
+	NBL_CONST_REF_ARG(pair<KeyType, ValueType>) partnerHiPair,
+	NBL_CONST_REF_ARG(Comparator) comp)
+{
+	const bool loSelfSmaller = comp(loPair.first, partnerLoPair.first);
+	const bool takePartnerLo = takeLarger ? loSelfSmaller : !loSelfSmaller;
```

Comment:
there's a faster way to write this `loSelfSmaller==takeLarger`, look at a T/F table of your current expression

Reply:
are you sure this is correct?
https://github.com/Devsh-Graphics-Programming/Nabla/pull/940#discussion_r2411943639

if I myself am smaller, then `comp` which is `<` will return true, but  I will swap with partner when `takeLarger` is true

did you mess up variable naming here?